### DISCUSSION TITLE
Setting of "workflows.argoproj.io/completed": "true" not possible for pods?

### DISCUSSION THREAD

**Discussion by bananabob93** [argoproj/argo-workflows#10177] [Q&A]
Hello all,

I have a question regarding this the setting of the label "workflows.argoproj.io/completed": "true" after completion of  a workflow. I get the following error in my workflow controller (Pod created through a CronWorkflow):
`level=warning msg="failed to clean-up pod" action=labelPodCompleted error="Pod \"my-pod\" is invalid: spec: Forbidden: pod updates may not change fields other than spec.containers[*].image\, spec.initContainers[*].image, spec.activeDeadlineSeconds, spec.tolerations (only additions to existing tolerations) or spec.terminationGracePeriodSeconds`

As far as I know this is intended Kuberntes behaviour - Deployments can be updated and create a new pod, but not a pod directly?
But since I haven't found any other issues discussing this error - Might this be an error related to my specific kubernetes configuration? 
Thank you!

**Comment by sarabala1979** [comment]
looks like you have  RBAC pod fields. The controller needs to update the pod's labels. If you add pod.metadata.label on your role. it will work.

**Reply by bananabob93** [reply]
Hello, and thanks for the quick reply. 

I'm not aware of a resource "pod.metadata.label" for a role - I do have the following on my role though:

```
  resources:
  - pods
  - pods/exec
  verbs:
  - create
  - get
  - list
  - watch
  - update
  - patch
  - delete
```
I thought that should suffice?

**Reply by bananabob93** [reply]
I tested updating the role as described, but I still see the same error in the workflow controller. 

```
    resources:
      - pods
      - pods/exec
      - pod.metadata.label
    verbs:
      - create
      - get
      - list
      - watch
      - update
      - patch
      - delete
```

**Reply by bananabob93** [reply]
As suspected it did not have anything to do with the role. If I deploy argo workflows with the same configuration in a blank cluster everything works as expected. But together with kyverno policies in the cluster, the updating of the label does not work because kyverno is apparently checking the spec against the kube-apiserver. I initially suspected that it has to do with the controller trying to update the pod and not the ownerReference, which is the workflow, but the workflow is updated and successfully has the label set to "true".
